docs(sparkplug): add Sparkplug Device documentation (#5108)
eclipse/kura: docs/cloud-platform/kura-sparkplug.md

### Cloud Endpoint Layer Configuration

The cloud endpoint layer allows to attach [`CloudPublisher`](https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloudconnection/publisher/CloudPublisher.java)s and [`CloudSubscriber`](https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloudconnection/subscriber/CloudSubscriber.java)s to publish/subscribe messages on Sparkplug topics.

#### Sparkplug Device

> Each [`CloudPublisher`](https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloudconnection/publisher/CloudPublisher.java) attached to this cloud connection acts as a **Sparkplug Device**. The configuration for it is shown in the picture below.

Reviewer: ```suggestion
Each [`CloudPublisher`](https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloudconnection/publisher/CloudPublisher.java) attached to this cloud connection acts as a **Sparkplug Device**. The corresponding configuration is shown in the picture below.
```

![](./images/sparkplugCloudPublisher.png)

> The parameter specified as `device.id` will dictate the Sparkplug device identifier used to publish messages from this cloud publisher. A device `DBIRTH` message is sent from this publisher when the first publish occurs or when a the set of published metrics is changed. Subsequent publishings will be done on the Sparkplug topic to publish device data (`DDATA` message type).

Reviewer: ```suggestion
The parameter specified as `device.id` will dictate the Sparkplug device identifier used to publish messages from this cloud publisher. A device `DBIRTH` message is immediately sent when the first publish occurs or when the set of published metrics is changed. The subsequent messages will be published as Sparkplug device data (`DDATA` message type).
```

> The Sparkplug Device implemented by this publisher **does not implement** the following standard optional specifications:

Reviewer: ```suggestion
The Sparkplug Device implemented by this publisher **does not support** the following features (optional in the Eclipse Sparkplug specification):
```

    | Java Type | Sparkplug Type |
    | - | - |
    | `Boolean` | `DataType.Boolean` |
    | `byte[]` | `DataType.Bytes` |
    | `Double` | `DataType.Double` |
    | `Float` | `DataType.Float` |
    | `Byte` | `DataType.Int8` |
    | `Short` | `DataType.Int16` |
    | `Integer` | `DataType.Int32` |
    | `Long` | `DataType.Int64` |
    | `String` | `DataType.String` |
    | `Date` | `DataType.DateTime` |
    | `BigInteger` | `DataType.UInt64` |

    All other Java types will cause the application to throw an Exception.

- [`KuraPayload.getBody()`](https://github.com/eclipse/kura/blob/d53ec833b7438a70a0e3a79406f4c8aed52e94f0/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L120), if non null, will be copied into the **body** of the [Sparkplug payload](https://github.com/eclipse-sparkplug/sparkplug/blob/3.x/specification/src/main/asciidoc/chapters/Sparkplug_6_Payloads.adoc#payload)

> - [`KuraPayload.getPosition()`](https://github.com/eclipse/kura/blob/d53ec833b7438a70a0e3a79406f4c8aed52e94f0/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L84), if non  null, will be used to create the following metrics from the [`KuraPosition`](https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPosition.java#L28) object is the value in there is not null (with the corresponding Sparkplug data types):

Reviewer: ```suggestion
- [`KuraPayload.getPosition()`](https://github.com/eclipse/kura/blob/d53ec833b7438a70a0e3a79406f4c8aed52e94f0/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L84), if not null, will be used to create the following metrics from the [`KuraPosition`](https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPosition.java#L28) object, if the value in there is not null (with the corresponding Sparkplug data types):
```

    - **kura.position.altitude**: `DataType.Double`
    - **kura.position.heading**: `DataType.Double`
    - **kura.position.latitude**: `DataType.Double`
    - **kura.position.longitude**: `DataType.Double`
    - **kura.position.precision**: `DataType.Double`
    - **kura.position.satellites**: `DataType.Int32`
    - **kura.position.status** `DataType.Int32`
    - **kura.position.speed**: `DataType.Double`
    - **kura.position.timestamp**: `DataType.DateTime`

> - [`KuraPayload.getTimestamp()`](https://github.com/eclipse/kura/blob/d53ec833b7438a70a0e3a79406f4c8aed52e94f0/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L76), if non null, will be used as the timestamp metric of the Sparkplug payload

Reviewer: ```suggestion
- [`KuraPayload.getTimestamp()`](https://github.com/eclipse/kura/blob/d53ec833b7438a70a0e3a79406f4c8aed52e94f0/kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/message/KuraPayload.java#L76), if not null, it will be used as the timestamp metric of the Sparkplug payload
```
